docs/src/pages/reference/configuration-reference.md

```diff
@@ -3,22 +3,44 @@ layout: ~/layouts/MainLayout.astro
 title: Configuration Reference
 ---
 
-To configure Astro, add an `astro.config.mjs` file in the root of your project. All settings are optional.
-
-You can view the full configuration API (including information about default configuration) on [GitHub.](https://github.com/withastro/astro/blob/latest/packages/astro/src/%40types/astro.ts)
+To configure Astro, add an `astro.config.mjs` file to the root of your project.
 
 ```js
-// Example: astro.config.mjs
-
-// @type-check enabled!
-// VSCode and other TypeScript-enabled text editors will provide auto-completion,
-// helpful tooltips, and warnings if your exported object is invalid.
-// You can disable this by removing "@ts-check" and `@type` comments below.
-
-// @ts-check
-export default /** @type {import('astro').AstroUserConfig} */ (
-  {
-    // ...
-  }
-);
+export default /** @type {import('astro').AstroUserConfig} */ ({
+  projectRoot: './',
+  public: './public/',
+  dist: './dist/',
+  src: './src/',
+  pages: './pages/',
+  renderers: [
+    '@astrojs/renderer-svelte',
+    '@astrojs/renderer-vue',
+    '@astrojs/renderer-react',
+    '@astrojs/renderer-preact',
+  ],
+});
 ```
+
+All settings are optional.
+
+The `projectRoot` option sets the working directory used by Astro. All other paths will be resolved from this. By default, this is the current working directory.
+
+The `public` option sets the directory used to resolve public assets. By default, this is the `public` directory.
+
+The `dist` option sets the directory used to write the production build of the project.
+
+The `src` option sets the directory used to resolve source files, like `pages`. By default, this is the `src` directory.
+
+The `pages` option sets the directory used to resolve pages, relative to the `src` option. By default, this is the `pages` directory.
+
+The `renderers` option assigns the framework renderers to be used by Astro. By default, Astro provides renderers for Svelte, Vue, React, and Preact. To assign no renderers at all, provide an empty array (`[]`).
+
+The `buildOptions` option configures your site, including the site URL (`buildOptions.site`), whether it should generate a sitemap (`buildOptions.sitemap`), and whether pages should be written as `path.html` or `path/index.html` (`buildOptions.pageUrlFormat`).
+
+The `devOptions` option configures your site for development, including the hostname (`devOptions.hostname`), the default port (`devOptions.port`), and whether urls should include a trailing slash (`devOptions.trailingSlash`).
+
+The `vite` option configures the internals of Vite. These options can be explored on [ViteJS.dev](https://vitejs.dev/config/).
```

Comment:
should we elaborate a bit more on this?
I mean we should prehaps say something along the lines of  "All configurations that are applied in vite are applied to Astro, these allow for modification and fine adjustments over how vite is configured for the project"

Reply:
I’m not ready to add that, yet. That’s my own limitation. I’m actually not sure if it’s accurate that all Vite configurations are applied to Astro. Is it okay to punt on this for now?